app/models/developer_key.rb

```diff
@@ -101,6 +101,14 @@ def account_name
     account.try(:name)
   end
 
+  def access_token_count
+    self.access_tokens.count
+  end
+
+  def last_used_at
+    self.access_tokens.order(last_used_at: :desc).first&.last_used_at
```

Comment:
Just a couple clarifications:Postgres likely wouldn't do a table scan since it already has the index